TFM does not build when enable TFM_PARTITION_FIRMWARE_UPDATE=y

**Describe the bug**
When try to build Zephyr v3.4.0-RC-1 with TFM and TFM_PARTITION_FIRMWARE_UPDATE=y the build fails.

```shell
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/gfbudke/zephyr/mainline/zephyr/samples/tfm_integration/tfm_secure_partition
-- CMake version: 3.22.1
-- Found Python3: /home/gfbudke/zephyr/mainline/zephyr/.venv-3.11/bin/python3.11 (found suitable exact version "3.11.2") found components: Interpreter 
-- Cache files will be written to: /home/gfbudke/.cache/zephyr
-- Zephyr version: 3.4.0-rc1 (/home/gfbudke/zephyr/mainline/zephyr)
-- Found west (found suitable version "1.0.0", minimum required is "0.14.0")
-- Board: b_u585i_iot02a_ns
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.1 (/home/gfbudke/zephyr-sdk-0.16.1)
-- Found toolchain: zephyr 0.16.1 (/home/gfbudke/zephyr-sdk-0.16.1)
-- Found Dtc: /home/gfbudke/zephyr-sdk-0.16.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/gfbudke/zephyr/mainline/zephyr/boards/arm/b_u585i_iot02a/b_u585i_iot02a_ns.dts
-- Generated zephyr.dts: /home/gfbudke/zephyr/mainline/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/gfbudke/zephyr/mainline/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/gfbudke/zephyr/mainline/zephyr/build/zephyr/dts.cmake
/home/gfbudke/zephyr/mainline/zephyr/build/zephyr/zephyr.dts:737.30-766.5: Warning (spi_bus_bridge): /soc/octospi@420d2400: node name for SPI buses should be 'spi'
<stdout>: Warning (spi_bus_reg): Failed prerequisite 'spi_bus_bridge'
Parsing /home/gfbudke/zephyr/mainline/zephyr/Kconfig
Loaded configuration '/home/gfbudke/zephyr/mainline/zephyr/boards/arm/b_u585i_iot02a/b_u585i_iot02a_ns_defconfig'
Merged configuration '/home/gfbudke/zephyr/mainline/zephyr/samples/tfm_integration/tfm_secure_partition/prj.conf'
Configuration saved to '/home/gfbudke/zephyr/mainline/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/gfbudke/zephyr/mainline/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /home/gfbudke/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/gfbudke/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
CMake Error at /home/gfbudke/zephyr/mainline/zephyr/cmake/modules/extensions.cmake:424 (add_library):
  Cannot find source file:

    /home/gfbudke/zephyr/mainline/zephyr/build/tfm/install/interface/src/tfm_fwu_api.c
Call Stack (most recent call first):
  /home/gfbudke/zephyr/mainline/zephyr/modules/trusted-firmware-m/CMakeLists.txt:382 (zephyr_library_named)


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
FATAL ERROR: command exited with status 1: /home/gfbudke/.local/bin/cmake -DWEST_PYTHON=/home/gfbudke/zephyr/mainline/zephyr/.venv-3.11/bin/python3 -B/home/gfbudke/zephyr/mainline/zephyr/build -GNinja -DBOARD=b_u585i_iot02a_ns -S/home/gfbudke/zephyr/mainline/zephyr/samples/tfm_integration/tfm_secure_partition
```

**To Reproduce**
1. Add below patch in the sample `samples/tfm_integration/tfm_secure_partition`
```diff
diff --git a/samples/tfm_integration/tfm_secure_partition/prj.conf b/samples/tfm_integration/tfm_secure_partition/prj.conf
index 2a97af0cb0..8bd2b2899b 100644
--- a/samples/tfm_integration/tfm_secure_partition/prj.conf
+++ b/samples/tfm_integration/tfm_secure_partition/prj.conf
@@ -7,3 +7,4 @@
 CONFIG_BUILD_WITH_TFM=y
 CONFIG_TFM_PROFILE_TYPE_NOT_SET=y
 CONFIG_TFM_CONNECTION_BASED_SERVICE_API=y
+CONFIG_TFM_PARTITION_FIRMWARE_UPDATE=y
```
2. Try to build using:
`west build -b b_u585i_iot02a_ns samples/tfm_integration/tfm_secure_partition`
3. See the error

**Expected behavior**
The TFM should configure and build correctly when selecting secure partitions.

**Impact**
This is a regression with v3.2, possible v3.3

**Environment (please complete the following information):**
 - Platform: b_u585i_iot02a_ns
 - OS: Linux
 - Toolchain Zephyr SDK 0.16.1
 - Commit v3.4.0-RC1

CC: @erwango

--- a/modules/trusted-firmware-m/CMakeLists.txt
+++ b/modules/trusted-firmware-m/CMakeLists.txt
@@ -209,6 +209,7 @@ if (CONFIG_BUILD_WITH_TFM)
 
     ${TFM_INTERFACE_SOURCE_DIR}/tfm_attest_api.c
     ${TFM_INTERFACE_SOURCE_DIR}/tfm_crypto_api.c
+    ${TFM_INTERFACE_SOURCE_DIR}/tfm_fwu_api.c
     ${TFM_INTERFACE_SOURCE_DIR}/tfm_its_api.c
     ${TFM_INTERFACE_SOURCE_DIR}/tfm_platform_api.c
     ${TFM_INTERFACE_SOURCE_DIR}/tfm_ps_api.c

